internal/api/verify.go

```diff
@@ -364,13 +365,29 @@ func (a *API) prepErrorRedirectURL(err *HTTPError, w http.ResponseWriter, r *htt
 	}
 	q.Set("error_code", strconv.Itoa(err.Code))
 	q.Set("error_description", err.Message)
-	return rurl + "#" + q.Encode()
+	if flowType == models.PKCEFlow {
+		fragmentDelimiter = "?"
+		// For backward compatibility with client library, we append implict error redirect as well
+		return a.prepErrorRedirectURL(err, w, r, rurl+fragmentDelimiter+q.Encode(), models.ImplicitFlow)
+	} else {
+		fragmentDelimiter = "#"
+	}
+	return rurl + fragmentDelimiter + q.Encode()
 }
 
-func (a *API) prepRedirectURL(message string, rurl string) string {
+func (a *API) prepRedirectURL(message string, rurl string, flowType models.FlowType) string {
 	q := url.Values{}
 	q.Set("message", message)
-	return rurl + "#" + q.Encode()
+	var fragmentDelimiter string
+
+	if flowType == models.PKCEFlow {
+		fragmentDelimiter = "?"
+		// For backward compatibility with client library, we append implict error redirect as well
+		return a.prepRedirectURL(message, rurl+fragmentDelimiter+q.Encode(), models.ImplicitFlow)
```

Comment:
These lines:

```
		return a.prepRedirectURL(message, rurl+fragmentDelimiter+q.Encode(), models.ImplicitFlow)
```

Seem to repeat a bit? Maybe extract them into a function we can test?